code/game/objects/structures/displaycase.dm

```diff
@@ -106,13 +106,25 @@
 	return
 
 /obj/structure/displaycase/attackby(obj/item/weapon/W, mob/user, params)
-	if(W.GetID() && electronics && !broken)
+	if(W.GetID() && !broken)
 		if(allowed(user))
 			user <<  "<span class='notice'>You [open ? "close":"open"] the [src]</span>"
-			open = !open
-			update_icon()
+			toggle_lock(user)
 		else
 			user <<  "<span class='warning'>Access denied.</span>"
+	else if(istype(W, /obj/item/weapon/weldingtool) && user.a_intent == "help" && !broken)
+		var/obj/item/weapon/weldingtool/WT = W
+		if(obj_integrity < max_integrity && WT.remove_fuel(0, user))

```

Comment:
Thanks for pointing that out. Also, it's interesting to note that a bunch of objects don't use any fuel. Most notably, windows don't use up any fuel when they're being repaired.